ldms/src/sampler/llnl/edac.c

```diff
@@ -390,15 +390,20 @@ static int sample(struct ldmsd_sampler *self)
 		myFile = fopen(command[i], "r");
 		if (myFile == NULL)
 		{
-			msglog(LDMSD_LERROR, SAMP ": failed to open file\n");
+			msglog(LDMSD_LERROR, SAMP ": failed to open file %s\n",
+					command[i]);
 			rc = EINVAL;
 			edac_valid=0;
 			goto out;
 		}
+		errno = 0;
```

Comment:
Why are you setting errno to zero here?

Reply:
that's a good question. it should be higher up since (at least per man) only sscanf and fopen use errno rather than fgets. will revisit all the error reports here since they are mushing all cases in to einval.